### PR TITLE
fix(instr-express): normalize paths with double slashes

### LINKED ISSUE #1547: Express instrumentation does not normalize paths (removing double slashes)
### What version of OpenTelemetry are you using?

auto-instrumentations-node 0.37.1
instrumentation-express 0.32.4

### What version of Node are you using?

18

### What did you do?

define some routes like this:

```
server.use(`/foo/`,
router.use('/v5/',
router.use(
    '/bar/',
router.get('/refresh',
```

### What did you expect to see?

http.route attribute in traces should contain:

```
/foo/v5/bar/refresh
```

Instrumentation should normalise the path in the same way that Express does (Express knows that declaring /v5/ inside /foo/ means /foo/v5/ not /foo//v5/ )

### What did you see instead?

```
/foo//v5//bar//refresh
```

### Additional context

express 4.18.2

### PR DESCRIPTION
**Pull request by JamieDanielson**
## Which problem is this PR solving?

- closes #1547 

## Short description of the changes

- Add replace to remove duplicate slashes that may exist in defined routes
- Add unit test confirming route does not contain extra slash
- Add route to util `serverWithMiddleware` that contains double slashes, for use in test

Test results before the change:

```
+ expected - actual
-/double-slashes//:id
+/double-slashes/:id
```

### CODE REVIEW

File: plugins/node/opentelemetry-instrumentation-express/test/express.test.ts [hunk 1]
```diff
@@ -455,6 +455,43 @@ describe('ExpressInstrumentation', () => {
...
+      const rootSpan = tracer.startSpan('rootSpan');
+      let finishListenerCount: number | undefined;
+      let rpcMetadata: RPCMetadata | undefined;
+      const httpServer = await serverWithMiddleware(tracer, rootSpan, app => {
+        app.use(express.json());
+        app.use((req, res, next) => {
+          rpcMetadata = getRPCMetadata(context.active());
+          res.on('finish', () => {
+            finishListenerCount = res.listenerCount('finish');
+          });
```
Comment (pichlermarc) [inline plugins/node/opentelemetry-instrumentation-express/test/express.test.ts:468]: Hmm, is there a specific goal with this and the associated assertion below? :thinking: 
To my untrained eye it seems somewhat unrelated to what's being tested here.

**Review by pichlermarc — COMMENTED** [review]
Looks good already, thanks for working on this :slightly_smiling_face: 
Just one question about the added test

File: plugins/node/opentelemetry-instrumentation-express/test/express.test.ts (hunk 1 above)
Comment (JamieDanielson) [reply plugins/node/opentelemetry-instrumentation-express/test/express.test.ts]: Oh good point, looks like extra copy-pasta leftover from another test. Removed

**Review by pichlermarc — APPROVED** [review]
Looks good, thanks for working on this :slightly_smiling_face:
